packages/salesforcedx-vscode-core/src/commands/util/parameterGatherers.ts

```diff
@@ -116,12 +116,24 @@ export class FileSelector implements ParametersGatherer<FileSelection> {
 
 export class SelectFileName
   implements ParametersGatherer<{ fileName: string }> {
+  private validator?: MaxLengthValidator;
+
+  constructor(validator?: MaxLengthValidator) {
+    this.validator = validator;
+  }
+
   public async gather(): Promise<
     CancelResponse | ContinueResponse<{ fileName: string }>
   > {
     const fileNameInputOptions = {
-      prompt: nls.localize('parameter_gatherer_enter_file_name')
+      prompt: nls.localize('parameter_gatherer_enter_file_name'),
+      ...(this.validator && {
+        validateInput: text => {
+          return this.validator?.validate(text);
+        }
+      })
```

Comment:
optionally added this validator to the input so that other instances of `SelectFileName` could still work as they were before